bouture.js

```diff
@@ -57,32 +58,63 @@ function completeElement (tag, args) {
           break
         }
         Object.keys(arg)
-          .forEach(attributeName => {
-            const attributeValue = arg[attributeName]
-            switch (typeof attributeValue) {
-              case 'boolean':
-                if (attributeValue) {
-                  element.setAttribute(attributeName, '')
-                }
-                break
-              case 'number':
-                if (!Number.isNaN(attributeValue)) {
-                  element.setAttribute(attributeName, attributeValue)
+          .forEach(keyName => {
+            function getEventName (name) {
+              return name.replace(/^once|on/, '').toLowerCase()
+            }
+
+            function isEvent (name) {
+              if (name.match(/^on[A-Z]|once[A-Z]/)) {
+                return EVENTNAMES.has(getEventName(name))
+              }
+            }
+
+            function getEventOrAttributeType (name) {
+              if (isEvent(name)) {
+                return 'event'
+              } else {
+                return 'attribute'
+              }
+            }
+
+            switch (getEventOrAttributeType(keyName)) {
```

Comment:
I'm surprised by this `switch`. Why not just `if (isEvent(keyName))`?

Reply:
Before your remark, I thought it was more explicit ;)